runtime/parachains/src/paras_inherent.rs

```diff
@@ -252,8 +251,15 @@ pub mod pallet {
 				parent_header,
 				disputes,
 			} = data;
-			let disputes_len = disputes.len() as u32;
-			let bitfields_len = signed_bitfields.len() as u32;
+			let total_weight = paras_inherent_total_weight::<T>(
+				&backed_candidates,
+				&signed_bitfields,
+				&disputes,
+			);
+
+			if total_weight > <T as frame_system::Config>::BlockWeights::get().max_block {
+				return Ok(Some(minimal_inherent_weight::<T>()).into());
+			}
```

Comment:
This might not be necessary. I think the weight annotations above the functions would keep it from getting queued if overweight.

Reply:
TODO: Make sure we can remove